mirgecom/materials/tacot.py

```diff
@@ -0,0 +1,527 @@
+                0.0, (
+                    -(30.*((chi[0] - 0.00)/30.)**3)*12000.
+                    * actx.np.exp(-8556.000/temperature))),
+            actx.np.where(actx.np.less(temperature, self._Tcrit[1]),
+            # reaction 2
+            0.0, (
+                -(90.*((chi[1] - 60.0)/90.)**3)*4.48e9
+                * actx.np.exp(-20444.44/temperature))),
+            # fiber oxidation: include in the RHS but dont do anything with it.
+            temperature*0.0])
+
+
+def eval_spline(x, x_bnds, coeffs) -> DOFArray:
+    r"""Evaluate spline $a(x-x_i)^3 + b(x-x_i)^2 + c(x-x_i) + d$.
+
+    Parameters
+    ----------
+    x: :class:`~meshmode.dof_array.DOFArray`
+        The value where $f(x)$ will be evaluated.
+    x_bnds: :class:`numpy.ndarray` with shape ``(m,)``
+        The $m$ nodes $x_i$ for the different segments of the spline.
+    coeffs: :class:`numpy.ndarray` with shape ``(4,m)``
+        The 4 coefficients for each segment $i$ of the spline.
+    """
+    actx = x.array_context
+
+    val = x*0.0
+    for i in range(0, len(x_bnds)-1):
+        val = (
+            actx.np.where(actx.np.less(x, x_bnds[i+1]),
+            actx.np.where(actx.np.greater_equal(x, x_bnds[i]),
+                coeffs[0, i]*(x-x_bnds[i])**3 + coeffs[1, i]*(x-x_bnds[i])**2
+                + coeffs[2, i]*(x-x_bnds[i]) + coeffs[3, i], 0.0), 0.0)) + val
+
+    return val
+
+
+def eval_spline_derivative(x, x_bnds, coeffs) -> DOFArray:
+    """Evaluate analytical derivative of a spline $3a(x-x_i)^2 + 2b(x-x_i) + c$.
+
+    Parameters
+    ----------
+    x: :class:`~meshmode.dof_array.DOFArray`
+        The value where $f(x)$ will be evaluatead.
+    x_bnds: :class:`numpy.ndarray` with shape ``(m,)``
+        The $m$ nodes $x_i$ for the different segments of the spline.
+    coeffs: :class:`numpy.ndarray` with shape ``(4,m)``
+        The 4 coefficients for each segment $i$ of the spline.
+    """
+    actx = x.array_context
+
+    val = x*0.0
+    for i in range(0, len(x_bnds)-1):
+        val = (
+            actx.np.where(actx.np.less(x, x_bnds[i+1]),
+            actx.np.where(actx.np.greater_equal(x, x_bnds[i]),
+                3.0*coeffs[0, i]*(x-x_bnds[i])**2 + 2.0*coeffs[1, i]*(x-x_bnds[i])
+                + coeffs[2, i], 0.0), 0.0)) + val
+
+    return val
+
+
+class GasProperties:
+    """Simplified model of the pyrolysis gas using tabulated data.
+
+    This section is to be used when species conservation is not employed and
+    the output gas is assumed to be in chemical equilibrium.
+    The table was extracted from the suplementar material from the
+    ablation workshop. Some lines were removed to reduce the number of spline
+    interpolation segments.
+
+    .. automethod:: gas_enthalpy
+    .. automethod:: gas_heat_capacity
+    .. automethod:: gas_molar_mass
+    .. automethod:: gas_dMdT
+    .. automethod:: gas_viscosity
+    .. automethod:: gas_thermal_conductivity
```

Comment:
Remove `gas_` prefixes to match `SolidProperties`?

Reply:
Done in [bc52690](https://github.com/illinois-ceesd/mirgecom/pull/875/commits/bc52690831096a0ecc59b84d502380dc28fdf6a9)